### DISCUSSION TITLE
A2DP Media Codec Selection Sound Sampling Bit Selection.

### DISCUSSION THREAD

**Discussion by yhtname** [pschatzmann/ESP32-A2DP#508] [Q&A] [answered]
Thanks to the work, I have created a Bluetooth audio decoder using the ESP-A2DP, which uses I2S for output.
The difficulty I'm having is that the audio decoding format is SBC 16 bit, which causes some loss of sound quality.
I would like to ask if there is any way to set it to SBC 24 bit?

In esp_a2dp_api.h you can also find

#define ESP_A2D_MCT_SBC (0) /*! < SBC */
#define ESP_A2D_MCT_M12 (0x01) /*! < MPEG-1, 2 Audio */ #define ESP_A2D_MCT_M12 (0x01) /*!
#define ESP_A2D_MCT_M24 (0x02) /*! < MPEG-2, 4 AAC */ #define ESP_A2D_MCT_M24 (0x02) /*!
#define ESP_A2D_MCT_ATRAC (0x04) /*! < ATRAC series */ #define ESP_A2D_MCT_ATRAC (0x04) /*!
#define ESP_A2D_MCT_NON_A2DP (0xff) /*!

Does this mean that the Bluetooth decoder supports AAC, and by what means is it set to AAC mode? This would give a big boost

Thanks for help!

**Comment by pschatzmann** [comment, marked as answer]
No, this is not supported by Espressif, so this is not available in Arduino!
